cpptod.dd

```diff
@@ -829,6 +830,86 @@ void test()
 }
 ------
 
+$(HR)
+
+$(H3 $(DDOC_ANCHOR interfaces)Interfaces)
+
+Interfaces in C++ are used to describe the behavior or capabilities of a class without
+commiting to a particular implementation of the class.
+
+$(H4 Interfaces in C++)
+
+  "C++ interfaces are implemented by means of $(I abstract classes). An abstract class
+	defines interface methods as pure virtual functions. A class implementing the interface
+	inherits the abstract class and implements its pure virtual functions."
+
+$(CPPCODE
+class Interface {
+  public:
+    virtual int function() = 0;
+};
+
+class FirstVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+
+class SecondVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+)
+
+The interface is used as follows:
+
+$(CPPCODE
+FirstVariantOfImplementation FirstVariant;
```

Comment:
Not really a big deal, but these names are tediously long. I think `FirstVariant` and `SecondVariant` would suffice.